akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala

```diff
@@ -79,7 +79,7 @@ private[http] class HttpResponseParser(protected val settings: ParserSettings, p
               val reason = asciiString(input, reasonStartIdx, reasonEndIdx)
               StatusCodes.custom(code, reason)
             } catch {
-              case _: Exception => badStatusCodeSpecific(code)
+              case NonFatal(_) => badStatusCodeSpecific(code)
```

Comment:
There are no fatals that are `Exception` they are all `Throwable` or `Error`.

Reply:
I agree, though, I'm also fine with using the common style of using `NonFatal` here. It's probably slightly slower (since it checks for more cases) but that shouldn't matter since it is an error case anyway.